beacon-chain/db/kv/blocks.go

```diff
@@ -549,11 +551,28 @@ func (s *Store) RegistrationByValidatorID(ctx context.Context, id primitives.Val
 		if enc == nil {
 			return errors.Wrapf(ErrNotFoundFeeRecipient, "validator id %d", id)
 		}
-		return decode(ctx, enc, reg)
+		if err := decode(ctx, enc, reg); err != nil {
+			return err
+		}
+		if timeStampExpired(reg.Timestamp) {
+			// don't delete in bucket, db will eventually be removed.
+			return errors.Wrapf(ErrNotFoundFeeRecipient, "validator id %d", id)
+		}
+		return nil
 	})
 	return reg, err
 }
 
+func timeStampExpired(ts uint64) bool {
+	expiryDuration := time.Duration(params.BeaconConfig().SecondsPerSlot*uint64(params.BeaconConfig().SlotsPerEpoch)*3) * time.Second
+	// safely convert unint64 to int64
+	t, err := strconv.ParseInt(fmt.Sprint(ts), 10, 64)
+	if err != nil {
+		return false
+	}
+	return time.Unix(t, 0).Add(expiryDuration).Unix() < time.Now().Unix()
+}
```

Comment:
Is this different from the other timeStampExpired in this PR?

Reply:
yes this is the same, I didn't find a great place to place this but will try to put it in a place where it'd make sense. was thinking maybe we can keep it in both places until full removal but will try to find a better spot without circular dependencies.